src/serializer/ResidualHeapSerializer.js

```diff
@@ -1117,9 +1126,106 @@ export class ResidualHeapSerializer {
         undelay();
       });
     }
+    let classProperties = instance.classProperties;
+
+    if (classProperties !== undefined) {
+      // handle the constructor for the class
+      if (classProperties.methodType === "constructor") {
+        invariant(val.$HomeObject instanceof ObjectValue);
+        let properties = new Map(val.properties);
+
+        // we visit all of these as these, as they form part of the constructor for the class
+        // but we want to serialize them out as class methods in the ResidualFunctions
+        // rather than via the typical Prepack route of serializing properties
+        for (let [key] of val.properties) {
+          if (!this.residualHeapInspector.canIgnoreProperty(val, key)) {
+            let propertyValue = Get(this.realm, val, key);
+
+            if (key === "arguments" || key === "length" || key === "name" || key === "caller" || key === "prototype") {
+              properties.delete(key);
+              this.serializedValues.add(propertyValue);
```

Comment:
Prefer changing the visitor to understand classes to explicitly adding to serializedValues. (make a new visitValueClass method that visitValue or visitValueFunction calls to when it detects a class that decides not to visit the arguments, length, name etc).

Reply:
I can make this change, but I remember running into issues when I didn't visit `arguments` and `caller` (from memory).

Reply:
@cblappert I don't think this is as simple as not visiting it. 

`_emitObjectProperties()` goes through all the properties on the class object and all its methods (as expected), including `arguments`, `length', `key' etc, regardless if they've been visited or not – and tries to serialize them all. I don't want `_emitObjectProperties` to serialize these values – because we create the AST for them with entirely separate logic.